substrate/frame/contracts-registry/src/migrations.rs

```diff
@@ -0,0 +1,98 @@
+use super::{pallet::Config, *};
+use frame_support::{pallet_prelude::*, storage_alias, traits::OnRuntimeUpgrade};
+use frame_system::pallet_prelude::BlockNumberFor;
+use sp_std::vec::Vec;
+
+#[cfg(feature = "try-runtime")]
+use sp_runtime::TryRuntimeError;
+
+pub mod v2 {
+	use super::*;
+
+	const TARGET: &'static str = "runtime::llm::migration::v2";
+    pub type I = ();
+
+	pub struct Migration<T, OldInterval>(sp_std::marker::PhantomData<(T, OldInterval)>);
+
+    #[storage_alias]
+    pub type PartiesSignatures<T: Config<I>> = StorageMap<
+        Pallet<T>,
+        Twox64Concat,
+        ContractIndex,
+        Vec<<T as frame_system::Config>::AccountId>,
+        OptionQuery,
+    >;
+	
+	#[storage_alias]
+	pub type JudgesSignatures<T: Config<I>> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		ContractIndex,
+		Vec<<T as frame_system::Config>::AccountId>,
+		OptionQuery,
+	>;
+
+	impl<T: Config, OldInterval: Get<BlockNumberFor<T>>> OnRuntimeUpgrade
+		for Migration<T, OldInterval>
+	{
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
+			assert!(StorageVersion::get::<Pallet<T>>() == 1, "can only upgrade from version 1");
+
+			Ok(().encode())
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			let weight = T::DbWeight::get().reads(1);
+			if StorageVersion::get::<Pallet<T>>() != 1 {
+				log::warn!(
+					target: TARGET,
+					"skipping on_runtime_upgrade: executed on wrong storage version.\
+				Expected version 1"
+				);
+				return weight;
+			}
+
+			let _ = v2::JudgesSignatures::<T>::clear(u32::MAX, None);
+			let _ = v2::PartiesSignatures::<T>::clear(u32::MAX, None);
+
+            Contracts::<T>::translate(|_, maybe_contract_data: Option<
```

Comment:
I'm like 90% sure `OldContractDataStorage` shouldn't be wrapped in an `Option<>` here. Are you sure about it? Does it work correctly in practice?